src/components/autocomplete/autocomplete-base-element.ts

```diff
@@ -0,0 +1,469 @@
+    const signal = this.abort.signal;
+    const formField = this.closest('sbb-form-field') ?? this.closest('[data-form-field]');
+
+    if (formField) {
+      this.negative = formField.hasAttribute('negative');
+    }
+
+    if (this._didLoad) {
+      this._componentSetup();
+    }
+    this.syncNegative();
+
+    this.addEventListener('click', (e: MouseEvent) => this.onOptionClick(e), { signal });
+  }
+
+  protected override willUpdate(changedProperties: PropertyValues<this>): void {
+    super.willUpdate(changedProperties);
+
+    if (changedProperties.has('origin')) {
+      this._resetOriginClickListener(this.origin, changedProperties.get('origin'));
+    }
+    if (changedProperties.has('trigger')) {
+      this._resetTriggerClickListener(this.trigger, changedProperties.get('trigger'));
+    }
+    if (changedProperties.has('negative')) {
+      this.syncNegative();
+    }
+  }
+
+  protected override firstUpdated(changedProperties: PropertyValues<this>): void {
+    super.firstUpdated(changedProperties);
+
+    this._componentSetup();
+    this._didLoad = true;
+  }
+
+  public override disconnectedCallback(): void {
+    super.disconnectedCallback();
+    this._triggerEventsController?.abort();
+    this._openPanelEventsController?.abort();
+  }
+
+  /** When an option is selected, update the input value and close the autocomplete. */
+  protected onOptionSelected(event: CustomEvent): void {
+    const target = event.target as SbbOptionBaseElement;
+    if (!target.selected) {
+      return;
+    }
+
+    // Deselect the previous options
+    this.options
+      .filter((option) => option.id !== target.id && option.selected)
+      .forEach((option) => (option.selected = false));
+
+    if (this.triggerElement) {
+      // Set the option value
+      this.triggerElement.value = target.value as string;
+
+      // Manually trigger the change events
+      this.triggerElement.dispatchEvent(new Event('change', { bubbles: true }));
+      this.triggerElement.dispatchEvent(new InputEvent('input', { bubbles: true, composed: true }));
+    }
+
+    this.close();
+  }
+
+  private _handleSlotchange(): void {
+    this._highlightOptions(this.triggerElement?.value);
+  }
+
+  /** The autocomplete should inherit 'readonly' state from the trigger. */
+  private get _readonly(): boolean {
+    return this.triggerElement?.hasAttribute('readonly') ?? false;
+  }
+
+  /** Removes trigger click listener on trigger change. */
+  private _resetOriginClickListener(
+    newValue?: string | HTMLElement,
+    oldValue?: string | HTMLElement,
+  ): void {
+    if (newValue !== oldValue) {
+      this._componentSetup();
+    }
+  }
+
+  /** Removes trigger click listener on trigger change. */
+  private _resetTriggerClickListener(
+    newValue?: string | HTMLElement,
+    oldValue?: string | HTMLElement,
+  ): void {
+    if (newValue !== oldValue) {
+      this._componentSetup();
+    }
+  }
+
+  private _componentSetup(): void {
+    if (!isBrowser()) {
```

Comment:
during rebase/merge main branch, you will see that isBrowser() was removed. please use !isServer (from lit)